model/jpa/src/main/java/org/keycloak/models/jpa/entities/OrganizationEntity.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.jpa.entities;
+
+import jakarta.persistence.Access;
+import jakarta.persistence.AccessType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.NamedQueries;
+import jakarta.persistence.NamedQuery;
+import jakarta.persistence.Table;
+
+@Table(name="ORGANIZATION")
+@Entity
+@NamedQueries({
+        @NamedQuery(name="deleteByRealm", query="delete from OrganizationEntity o where o.realm = :realm"),
+        @NamedQuery(name="getByRealm", query="select o.id from OrganizationEntity o where o.realm = :realm")
```

Comment:
Why there is queried IDs and not other fields?

I'd suggest to use `realmId` to make it clear it's indeed ID.
```suggestion
        @NamedQuery(name="deleteByRealm", query="delete from OrganizationEntity o where o.realmId = :realmId"),
        @NamedQuery(name="getByRealm", query="select o.id from OrganizationEntity o where o.realmId = :realmId")
```